pkg/querier/tripperware/query.proto

```diff
@@ -5,11 +5,44 @@ package tripperware;
 option go_package = "tripperware";
 
 import "gogoproto/gogo.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/any.proto";
 import "github.com/cortexproject/cortex/pkg/cortexpb/cortex.proto";
 
 option (gogoproto.marshaler_all) = true;
 option (gogoproto.unmarshaler_all) = true;
 
+message PrometheusResponse {
+  string Status = 1 [(gogoproto.jsontag) = "status"];
+  PrometheusData Data = 2 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "data,omitempty"];
+  string ErrorType = 3 [(gogoproto.jsontag) = "errorType,omitempty"];
+  string Error = 4 [(gogoproto.jsontag) = "error,omitempty"];
+  repeated tripperware.PrometheusResponseHeader Headers = 5 [(gogoproto.jsontag) = "-"];
+  repeated string Warnings = 6 [(gogoproto.jsontag) = "warnings,omitempty"];
+}
+
+message PrometheusData {
+  string ResultType = 1 [(gogoproto.jsontag) = "resultType"];
+  tripperware.PrometheusQueryResult Result = 2 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "result"];
```

Comment:
This is the main change. Previous [definition](https://github.com/cortexproject/cortex/blob/master/pkg/querier/tripperware/queryrange/queryrange.proto#L38) here uses `repated tripperware.SampleStream` which supports Matrix type only. This change makes it possible to support vector and other types so supporting both instant and range queries using the same type.

Reply:
Looks good to me. The new `PrometheusResponse` is pretty much the same as what we had for `PrometheusInstantQueryResponse` so should be easy to use for both types.